include/engine/plugins/plugin_base.hpp

```diff
@@ -179,18 +200,34 @@ class BasePlugin
             {
                 if (use_radiuses && parameters.radiuses[i])
                 {
-                    phantom_nodes[i] = facade.NearestPhantomNodes(parameters.coordinates[i],
-                                                                  number_of_results,
-                                                                  *parameters.radiuses[i],
-                                                                  parameters.bearings[i]->bearing,
-                                                                  parameters.bearings[i]->range);
+                    phantom_nodes[i] = facade.NearestPhantomNodes(
+                        parameters.coordinates[i],
+                        number_of_results,
+                        max_radius_when_bearings != EngineConfig::UNLIMITED
+                            ? std::min(*parameters.radiuses[i], max_radius_when_bearings)
+                            : *parameters.radiuses[i],
+                        parameters.bearings[i]->bearing,
+                        parameters.bearings[i]->range);
                 }
                 else
                 {
-                    phantom_nodes[i] = facade.NearestPhantomNodes(parameters.coordinates[i],
-                                                                  number_of_results,
-                                                                  parameters.bearings[i]->bearing,
-                                                                  parameters.bearings[i]->range);
+                    if (max_radius_when_bearings != EngineConfig::UNLIMITED)
+                    {
+                        phantom_nodes[i] =
```

Comment:
are these two functions actually different or use each other? I don't see why we split here in two functions.
In addition, why is `!= UNLIMITED` called without radius and `== UNLIMITED` called with radius?